src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -1108,6 +1109,58 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
       throw failures.failure(info, elementsShouldSatisfy(actual, unsatisfiedRequirements, info));
   }
 
+  /**
+   * Verifies that all given consumers can be satisfied by elements in the iterable under test with
+   * an element at most satisfying one consumer. No order requirement.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given {@code Iterable}.
+   * @param consumers the consumers that are expected to be satisfied by the elements of the given {@code Iterable}.
+   * @throws NullPointerException if the given consumers is {@code null}.
+   * @throws AssertionError if the given {@code Iterable} is {@code null}.
+   * @throws AssertionError if any {@code Consumer} in the consumers cannot be satisfied by elements in the given {@code Iterable}.
+   */
+  @SafeVarargs
+  public final <E> void assertSatisfy(AssertionInfo info, Iterable<? extends E> actual, Consumer<? super E>... consumers) {
+    assertNotNull(info, actual);
+    requireNonNull(consumers[0], "The Consumer<? super E>... expressing the assertions consumers must not be null");
+    List<E>[] satisfiedElementsLists = new ArrayList[consumers.length];
+    for (int i = 0; i < consumers.length; i++) {
+      Consumer<? super E> consumer = consumers[i];
+      satisfiedElementsLists[i] = stream(actual).filter(element -> {
+        try {
+          consumer.accept(element);
+        } catch (AssertionError ex) {
+          return false;
+        }
+        return true;
+      }).collect(toList());
+    }
+    if (!isSatisfied(satisfiedElementsLists, 0))
+      throw failures.failure(info, shouldSatisfy(actual));
+  }
+
+  private static <E> boolean isSatisfied(List<E>[] satisfiedElementsLists, int begin) {
+    // recursively test whether we can find any specific matching strategy that can meet the requirements
+    if (begin == satisfiedElementsLists.length) return true;   // all consumers have been satisfied
+    if (satisfiedElementsLists[begin].size() == 0) return false;   // no satisfied element for consumers[begin]
```

Comment:
Would ```isEmpty()``` be more readable here? 
```suggestion
    if (satisfiedElementsLists[begin].isEmpty()) return false;   // no satisfied element for consumers[begin]
```